src/vendor/CowWrapper.sol

```diff
@@ -0,0 +1,163 @@
+    /// @return true when prospectiveSolver is an authenticated solver, otherwise false.
+    function isSolver(address prospectiveSolver) external view returns (bool);
+}
+
+/// @title CoW Protocol Settlement Interface
+/// @notice Minimal interface for CoW Protocol's settlement contract
+/// @dev Used for type-safe calls to the settlement contract's settle function
+interface CowSettlement {
+    /// @notice Trade data structure matching GPv2Settlement
+    struct CowTradeData {
+        uint256 sellTokenIndex;
+        uint256 buyTokenIndex;
+        address receiver;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint32 validTo;
+        bytes32 appData;
+        uint256 feeAmount;
+        uint256 flags;
+        uint256 executedAmount;
+        bytes signature;
+    }
+
+    /// @notice Interaction data structure for pre/intra/post-settlement hooks
+    struct CowInteractionData {
+        address target;
+        uint256 value;
+        bytes callData;
+    }
+
+    /// @notice Returns the authentication contract used by the settlement contract.
+    function authenticator() external returns (CowAuthentication);
+
+    /// @notice Settles a batch of trades atomically
+    /// @param tokens Array of token addresses involved in the settlement
+    /// @param clearingPrices Array of clearing prices for each token
+    /// @param trades Array of trades to execute
+    /// @param interactions Array of three interaction arrays (pre, intra, post-settlement)
+    function settle(
+        address[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        CowTradeData[] calldata trades,
+        CowInteractionData[][3] calldata interactions
+    ) external;
+}
+
+/// @title CoW Protocol Wrapper Interface
+/// @notice Interface for wrapper contracts that add custom logic around CoW settlements
+/// @dev Wrappers can be chained together to compose multiple settlement operations
+interface ICowWrapper {
+    /// @notice A human readable label for this wrapper. Used for display in explorer/analysis UIs
+    function name() external view returns (string memory);
+
+    /// @notice Initiates a wrapped settlement call
+    /// @dev This is the entry point for wrapped settlements. The wrapper will execute custom logic
+    ///      before calling the next wrapper or settlement contract in the chain.
+    /// @param settleData ABI-encoded call to CowSettlement.settle()
+    /// @param wrapperData Encoded chain of wrapper-specific data followed by addresses of next wrappers/settlement
+    function wrappedSettle(bytes calldata settleData, bytes calldata wrapperData) external;
+}
+
+/// @title CoW Protocol Wrapper Base Contract
+/// @notice Abstract base contract for creating wrapper contracts around CoW Protocol settlements
+/// @dev A wrapper enables custom pre/post-settlement and context-setting logic and can be chained with other wrappers.
+///      Wrappers must:
+///      - Be approved by the CowAuthentication contract
```

Comment:
Link would be useful. Also, the name doens't match the contract. An alternative with a bit more context:

```suggestion
///      - Be an authorized solver (added as solver in `GPv2AllowListAuthentication` contract https://github.com/cowprotocol/contracts/blob/main/src/contracts/GPv2AllowListAuthentication.sol)
```